service/store/store.go

```diff
@@ -491,7 +494,13 @@ func (s *Store) WriteDataURI(bidID auction.BidID, payloadCid, uri string) (strin
 	if _, err := f.Seek(0, 0); err != nil {
 		return "", fmt.Errorf("seeking file to the beginning: %v", err)
 	}
-	ctx, cancel := context.WithTimeout(s.ctx, DataURIFetchTimeout)
+	fetchTimeout := DefaultDataURIFetchTimeout
+	if size != 0 {
+		// add baseline timeout for very small files
+		fetchTimeout = time.Minute + time.Duration(size/s.estDownloadSpeed)*time.Second
```

Comment:
I considered making the fetch timeout 2X the value, but don't want to introduce some magic number. The default is small enough, and the help message asked the miners to be conservative.